ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,507 @@
+                check["kw"]["suffix"] == "ca"
+
+    def test_source_ipa_roles_check_crlmanager(self):
+        """
+        This testcase checks the status of healthcheck tool
+        reflects correct information when crlgen is disabled
+        using ipa-crl-manage disable
+        """
+        self.master.run_command(["ipa-crlgen-manage", "disable"])
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.roles",
+                "--check",
+                "IPACRLManagerCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] == "True"
+
+    def test_ipa_healthcheck_remove(self):
+        """
+        This testcase checks the removal of of healthcheck tool
+        on replica and master
+        """
+        self.master.run_command(
+            ["dnf", "remove", "-y", "freeipa-healthcheck"])
+        self.replicas[0].run_command(
+            ["dnf", "remove", "-y", "freeipa-healthcheck"])
+
+
+class TestIpaHealthCheckWithoutDNS(IntegrationTest):
+    """
+    Test for ipa-healthcheck tool with IPA Master without DNS installed
+    """
+
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.uninstall_replica(cls.master, cls.replicas[0])
+        tasks.uninstall_master(cls.master)
+        tasks.install_master(cls.master, setup_dns=False)
+
+    def test_ipa_dns_systemrecords_check(self):
+        """
+        Test checks the result of IPADNSSystemRecordsCheck
+        when ipa-server is configured without DNS.
+        """
+        msg1 = "Expected SRV record missing"
+        msg2 = "Got {count} ipa-ca A records, expected {expected}"
+        tasks.install_package(self.master, HEALTHCHECK_PKG)
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.ipa.idns",
+                "--check",
+                "IPADNSSystemRecordsCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "WARNING"
+            assert check["kw"]["msg"] == msg1 or \
+                check["kw"]["msg"] == msg2
+        tasks.uninstall_master(self.master)
+
+
+class TestIpaHealthCheckWithKRA(IntegrationTest):
```

Comment:
This is already an extended test due to three separate server installations plus a replica. You could save an install by moving this into the first invocation and running ipa-kra-install instead as part of this test case.